public/app/features/explore/ExploreTimeControls.tsx

```diff
@@ -66,19 +69,25 @@ export class ExploreTimeControls extends Component<Props> {
     });
   };
 
+  changeTimeSync = () => {
+    const { onChangeTimeSync, exploreId } = this.props;
+    onChangeTimeSync(exploreId);
+  };
+
   render() {
-    const { range, timeZone } = this.props;
+    const { range, timeZone, splitted, syncedTimes } = this.props;
+    const syncButton = { onClick: this.changeTimeSync, synced: syncedTimes };
+    const timePickerCommonProps = {
+      value: range,
+      onChange: this.onChangeTimePicker,
+      timeZone,
+      onMoveBackward: this.onMoveBack,
+      onMoveForward: this.onMoveForward,
+      onZoom: this.onZoom,
+      selectOptions: this.setActiveTimeOption(defaultSelectOptions, range.raw),
+    };
+    const timePickerProps = splitted ? { syncButton } : null;
```

Comment:
Instead of passing syncButton props I suggest using composition and basically pass preconfigured SyncButton component as property. What I mean by that is interface like this:

```ts
interface TimePickerProps {
  ... // existing props
  syncButton?: React.DetailedReactHTMLElement<SyncButtonProps>
}
```

Thanks to that you can replace this in TimePicker:

```tsx
{syncButton && (
            <Tooltip content={syncTimesTooltip} placement="bottom">
              <button
                className={classNames('btn navbar-button navbar-button--attached', {
                  [`${styles.timePickerSynced}`]: syncButton.synced,
                })}
                onClick={() => syncButton.onClick()}
              >
                <i className="fa fa-link" />
              </button>
            </Tooltip>
          )}
```

with this 
```
{syncButton}
```

Reply:
Yes, that would be super clean. However,  I have 2 questions/points:
1. In ExploreTimeControls, we would need to add classNames, emotion and create "synced/asctive style" that has already been created in TimePicker for synced TimePicker part. 
2. We would need to add Tooltip there as well. 

Do you think it makes sense rather than having everything in TimePicker?

Reply:
I think all of the above would be implemented in `SyncButton` component.

So this component would have api like this: `<SyncButton onClick={[fn]} synced={[boolean]} />`.  

Then, this line:
```
const timePickerProps = splitted ? { syncButton } : null;
```
would become 
```
const syncButton = splitted ? <SyncButton onClick={this.changeTimeSync} synced={syncedTimes} /> : null;
```

And SyncButton implementation would be simply this code:
```tsx
<Tooltip content={syncTimesTooltip} placement="bottom">
  <button
    className={classNames('btn navbar-button navbar-button--attached', {
      [`${styles.timePickerSynced}`]: syncButton.synced,
    })}
    onClick={() => syncButton.onClick()}
  >
    <i className="fa fa-link" />
  </button>
</Tooltip>
```

IMO it's worth using composition rather than passing component specific props. Alternative would be to pass those sync button specific props(onClick, synced) implicitly rather than via `syncButton` object, like `onClick` would become `onSyncClick`, `synced` -> `isSynced`.